README.md

```diff
@@ -42,6 +42,14 @@ By default, the website will prompt you for a username and password to log in, b
 
 To start using your uploader with ShareX, go to your gallery page (which will be at the `base_url` you set in your configuration file) and click on the Share icon at the bottom of the page. This should automatically generate and download a ShareX configuration file. You can now open the file and ShareX should prompt to add your uploader.
 
+## ShareX File Name Templates
+
+The uploader supports using names generated by ShareX's templating system, which allows you to use information about this image to generate your file name. To enable this, you will need to set the `sharex_upload_naming_scheme` to `provided` in your `config.php`. 
+
+To customize the format of this name after adding your uploader to ShareX, you will need to open the `Custom uploader settings` from the `Destinations` menu in the ShareX application. Select the name of your uploader in the `Uploaders` menu. Now, you can edit the value of the  `name` field `Body` table on the right.
+
+This `name` field accepts any valid ShareX name template as a value. You can create a ShareX template string by opening the `Task Settings` menu from the ShareX application, opening the `File naming` tab, and selecting the `Name pattern for capture or clipboard upload` field. A dropdown will open on the right, which you can use to create the template string you want to use. You can then copy the string from this field back into the `name` field from the `Custom uploader settings`.
```

Comment:
I don't think adding this paragraph is necessary :P people figured it out before and its all explained within the shareX interface, I worry abit the readme is becoming insanely long

Reply:
From a beginner's point of view, it's nice having a fully explained method to do something. People might have figured it out, but there's likely a group of people that didn't know how to do it.

It might be better adding this as a wiki page instead, though.

Reply:
Alright, if someone else thinks its valuable thats good enough for me, lets leave it here for now and talk about moving things out of README.md in the future 👍